### PR TITLE
Custom JSON Coder for Postgres Coders

### PR DESCRIPTION
**Pull request by calebkleveter**
Relies on https://github.com/vapor/postgres-kit/pull/159

### CODE REVIEW

File: Sources/FluentPostgresDriver/PostgresRow+Database.swift [hunk 1]
```diff
@@ -1,13 +1,30 @@
-extension PostgresRow: DatabaseRow {
-    public func contains(field: String) -> Bool {
-        return self.column(field) != nil
+import class Foundation.JSONDecoder
+
+extension PostgresRow {
+    public func databaseRow(using decoder: PostgresDataDecoder) -> DatabaseRow {
```
Comment (tanner0101) [inline Sources/FluentPostgresDriver/PostgresRow+Database.swift]: We can keep these internal as well.

**Comment by calebkleveter** [comment]
What if we added a method to `DatabaseRow` that attempted to cast it to a `_PostgresDatabaseRow` and then extract the `PostgresRow` from that?

```swift
extension DatabaseRow {
    public func postgresRow() -> PostgresRow? {
        return (self as? _PostgresDatabaseRow)?.row
    }
}
```

File: Package.swift [hunk 2]
```diff
@@ -11,7 +11,7 @@ let package = Package(
     ],
     dependencies: [
         .package(url: "https://github.com/vapor/fluent-kit.git", from: "1.0.0-beta.2"),
-        .package(url: "https://github.com/vapor/postgres-kit.git", from: "2.0.0-beta.2"),
+        .package(url: "https://github.com/skelpo/postgres-kit.git", .branch("master")),
```
Comment (calebkleveter) [inline Package.swift]: We need to make sure this gets changes back to vapor/postgres-kit before it gets merged and tagged.

File: Sources/FluentPostgresDriver/PostgresRow+Database.swift [hunk 3]
```diff
@@ -1,13 +1,26 @@
-extension PostgresRow: DatabaseRow {
-    public func contains(field: String) -> Bool {
-        return self.column(field) != nil
+import class Foundation.JSONDecoder
+
+extension PostgresRow {
+    internal func databaseRow(using decoder: PostgresDataDecoder) -> DatabaseRow {
```
Comment (tanner0101) [inline Sources/FluentPostgresDriver/PostgresRow+Database.swift]: This should be `decoder:` to match but probably fine since it's internal anyway.

**Pull request merged**